### ISSUE TITLE
Documentation issue in go-app page

### ISSUE THREAD

**Issue by ricardoespsanto** [maxence-charriere/go-app#794]
How do I compose a web page? 

Say I want to have a menu, some element in the html body and an html footer.

I can't seem to wrap my head around composing components together to form a webpage.

`return app.Head().Body(&Menu{}).Div().Body(app.H1().Text("Welcome"))` 

If a component can only render independent of other components, how do we put them together? 

tx

**Comment by ricardoespsanto** [comment]
I have tried with html() but there's no Body() in it, I've been thinking that perhaps they could be multiple returns but that doesn't work either.. I'm a bit lost

**Comment by oderwat** [comment]
You create a hierarchy of components and use them inside each other. Look here https://go-app.dev/declarative-syntax for the "Nested Components" example.

I noticed that you are using `app.Head()` which most likely is not what you want to use. The page is already done when you render your component.

```
<html>
<head><-- data from app.Handler fields here --></head>
<body><--! your root component render ends up here--></body>
</html>
```

So, the basic structure of the page is already rendered when your root component (the one which starts nesting the other) is getting rendered. To change those, you need to change their field data in `app.Handler`.

Also, I am uncertain if you understood that the `Body()` you have in your code is not an HTML Body, but just "a collection of sub elements". So `app.Div().Body(app.H1().Text("overview"),app.P().Text("anything1"))` means:

```
<div>
<h1>overview</h1>
<p>anything</p>
</div>
```

I have no good example app in a public repo, but you could check out https://github.com/oderwat/go-guess-the-number-app/ which I wrote quite some time ago, as I started exploring go-app. I think we should make some of our internal, training repos available. Maybe in the new year, the team is mostly on vaccation already :)

**Comment by ricardoespsanto** [comment]
Thank you for your prompt reply! 

I copied the code in its worst possible state and I did not, realise that I couldn't just append stuff to the html Head object from my Render method. 

So, looking at your app, there's an append() method inside the Body of a Div. I'll look for this as I wasn't aware of it. 

But my problem still remains. If I can't create a root html element and then nest my elements in it then I will not be able to compose a webpage like:

`<html> 
<head /> 
<body>
<div><p>some content probably generated by some standalone element</p></div>
<footer>
  <p>some random text</p>
  <p><a href="mailto:me@myrandomdomain.com">Me@myRandomDomain.com</a></p>
</footer>
</html>`

my "root" element will always have to be something like a div. If that's the case even creating a menu from a simple CSS framework will be challenging as everything will be nested within that "root" div.

**Comment by oderwat** [comment]
> my "root" element will always have to be something like a div.

To have Go-App internal update mechanism to work, it needs to have a single root node for every component (at least how I understand the system). I think that this should probably being "body", but it is not. I needed to modify BODY and HTML, so I made a PR (#629) that lets me modify them (if needed).

Likewise, I have to confess that I never used a CSS framework that dictate me how a site must be structured. We usually use TailwindCSS (because that works as I expect a framework to work) and never run into a problem with that single root, besides on how the code locks.

Why can't you use `<main>` or the div as the root element? I do not see a problem there? Are there CSS Frameworks that work their way down on `<body>`?
